CMakeLists.txt

```diff
@@ -0,0 +1,478 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
+
+cmake_policy(SET CMP0077 NEW)
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+cmake_policy(SET CMP0126 NEW)
```

Comment:
I am using cmake 3.20.1 and getting warning:
```CMake Error at CMakeLists.txt:21 (cmake_policy):
  Policy "CMP0126" is not known to this version of CMake.
```

Reply:
`cmake --help-policy CMP0126` shows this policy was added in v3.21.

Reply:
Added version guards to avoid errors on earlier versions of CMake.